types/jest/index.d.ts

```diff
@@ -388,51 +386,277 @@ declare namespace jest {
         virtual?: boolean | undefined;
     }
 
-    type MockableFunction = (...args: any[]) => any;
-    type MethodKeysOf<T> = { [K in keyof T]: T[K] extends MockableFunction ? K : never }[keyof T];
-    type PropertyKeysOf<T> = { [K in keyof T]: T[K] extends MockableFunction ? never : K }[keyof T];
-    type ArgumentsOf<T> = T extends (...args: infer A) => any ? A : never;
+    interface ClassLike {
+        new (...args: any): any;
+    }
+    type FunctionLike = (...args: any) => any;
```

Comment:
Why not import these types from jest instead of copy'n'paste?

```
import type {
  ClassLike,
  FunctionLike,
  MockInstance as JestMockInstance,
  Mocked as JestMocked,
  MockedClass as JestMockedClass,
  MockedFunction as JestMockedFunction,
  MockedObject as JestMockedObject,
  SpyInstance as JestSpyInstance,
} from 'jest-mock';
```

https://github.com/renovatebot/renovate/pull/17737/files#diff-d28a461c46dd3d6440988800fbf174825ddbec8b60488095df612d33f1cad4df

Reply:
Thanks! That is also a possibility. This should fix build errors, but would still break `heft-jest`. Apparently they are shipping Jest 27, so to have a dependency from Jest 29 is odd in this case. Perhaps I could convince them to import from `jest-mock` v27 instead of depending on `@types/jest`? Could work. Thinking slowly how to put it all together.